internal/webhooks/test/clusterclass_test.go

```diff
@@ -483,3 +502,65 @@ func TestClusterClassWebhook_Delete_MultipleExistingClusters(t *testing.T) {
 	// Expect an error here as the webhook should not allow the deletion of an existing ClusterClass.
 	g.Expect(env.Delete(ctx, class)).To(Not(Succeed()))
 }
+
+// createTemplates builds and then creates all required ClusterClass templates in the envtest API server.
+func createTemplates(ns *corev1.Namespace) (func() error, error) {
+	// Templates for MachineInfrastructure, ClusterInfrastructure, ControlPlane and Bootstrap.
+	infrastructureMachineTemplate1 := builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureMachineTemplateName1).Build()
+	infrastructureMachineTemplate2 := builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureMachineTemplateName2).Build()
+	infrastructureClusterTemplate1 := builder.TestInfrastructureClusterTemplate(ns.Name, infrastructureClusterTemplateName1).Build()
+	infrastructureClusterTemplate2 := builder.TestInfrastructureClusterTemplate(ns.Name, infrastructureClusterTemplateName2).Build()
+	controlPlaneTemplate := builder.TestControlPlaneTemplate(ns.Name, controlPlaneTemplateName1).Build()
+	bootstrapTemplate := builder.TestBootstrapTemplate(ns.Name, bootstrapTemplateName1).Build()
+
+	// Create a set of templates from the objects above to add to the API server when the test environment starts.
+	initObjs := []client.Object{
+		infrastructureClusterTemplate1,
+		infrastructureClusterTemplate2,
+		infrastructureMachineTemplate1,
+		infrastructureMachineTemplate2,
+		bootstrapTemplate,
+		controlPlaneTemplate,
+	}
+	cleanup := func() error {
+		for _, o := range initObjs {
+			if err := env.CleanupAndWait(ctx, o); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	for _, obj := range initObjs {
+		if err := env.CreateAndWait(ctx, obj); err != nil {
+			return cleanup, err
+		}
+	}
+	return cleanup, nil
+}
+
+var (
+	workerClassName1 = "linux-worker"
```

Comment:
Should this lis of constants be at the top of the file? (nit: consider replacing var with const)

Reply:
While we're here, would be great if we can improve the const values, e.g. infracluster1 => infrastructureclustertemplate1,  infra1 => infrastructuremachinetemplate1, ...